### PR TITLE
feat(CMSIS,PeriphDrivers): Update SMON Libraries with latest naming convention and design implementation

### PR DESCRIPTION
**Pull request by sihyung-maxim**
### Description

- Update SMON library to use latest naming convention
- Adds latest SMON design implementation for MAX32572 and future parts going forward.
- Removes repetitive and unnecessary typecasts (and to match type definitions and usage of other libraries).

### Checklist Before Requesting Review

- [x] PR Title follows correct guidelines.
- [x] Description of changes and all other relevant information.
- [ ] (Optional) Link any related GitHub issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [ ] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

### CODE REVIEW

File: Libraries/PeriphDrivers/Include/MAX32520/smon.h [hunk 1]
```diff
@@ -47,82 +47,127 @@ extern "C" {
...
+    MXC_SMON_CLK_DIVIDE_1 = MXC_S_SMON_EXTSCN_DIVCLK_DIV1,
+    MXC_SMON_CLK_DIVIDE_2 = MXC_S_SMON_EXTSCN_DIVCLK_DIV2,
+    MXC_SMON_CLK_DIVIDE_4 = MXC_S_SMON_EXTSCN_DIVCLK_DIV4,
+    MXC_SMON_CLK_DIVIDE_8 = MXC_S_SMON_EXTSCN_DIVCLK_DIV8,
+    MXC_SMON_CLK_DIVIDE_16 = MXC_S_SMON_EXTSCN_DIVCLK_DIV16,
+    MXC_SMON_CLK_DIVIDE_32 = MXC_S_SMON_EXTSCN_DIVCLK_DIV32,
+    MXC_SMON_CLK_DIVIDE_64 = MXC_S_SMON_EXTSCN_DIVCLK_DIV64,
+
+    // Deprecated names
+    SMON_CLK_DIVIDE_1 = MXC_SMON_CLK_DIVIDE_1,
```
Comment (lorne-maxim) [inline Libraries/PeriphDrivers/Include/MAX32520/smon.h:70]: You can deprecate enums with an attribute tag.  For example:

SMON_CLK_DIVIDE_1 __attribute__ ((deprecated ("replaced by MXC_SMON_CLK_DIVIDE_1"))) = MXC_SMON_CLK_DIVIDE_1,

File: Libraries/PeriphDrivers/Source/SMON/smon_reva.h [hunk 2]
```diff
@@ -30,145 +30,61 @@
...
-typedef struct {
-    mxc_smon_reva_ext_sensor_t sensorNumber;
-    mxc_smon_reva_clk_divide_t clockDivide;
-    mxc_smon_reva_freq_divide_t freqDivide;
-    uint8_t errorCount;
-    uint8_t data;
-} mxc_smon_reva_ext_cfg_t;
-
-int MXC_SMON_RevA_ExtSensorEnable(mxc_smon_reva_regs_t *smon, mxc_smon_reva_ext_cfg_t *cfg,
+#if TARGET_NUM == 32572
```
Comment (lorne-maxim) [inline Libraries/PeriphDrivers/Source/SMON/smon_reva.h:33]: Is the #if necessary?  Can't smon_meXX.c files handle whether to call these functions or just return E_NOT_SUPPORTED?

File: Libraries/PeriphDrivers/Include/MAX32520/smon.h (hunk 1 above)
Comment (sihyung-maxim) [reply Libraries/PeriphDrivers/Include/MAX32520/smon.h:70]: Wow, didn't know this. This is a game-changer!

File: Libraries/PeriphDrivers/Source/SMON/smon_reva.h (hunk 2 above)
Comment (sihyung-maxim) [reply Libraries/PeriphDrivers/Source/SMON/smon_reva.h]: Nope, you're right. Originally, the new structs in the ME55 weren't added in the other parts yet which caused build errors for the ES17 and ME13A.

**Pull request closed without merging**